r/tests/testthat/test-schema.R

```diff
@@ -48,6 +47,11 @@ test_that("Schema $GetFieldByName", {
 })
 
 test_that("Schema extract (returns Field)", {
+  # TODO: should this return a Field or the Type?
```

Comment:
I think Field is right here, but I can see the other side too. But especially since names can be duplicated, I think it's easier to interact with them that way. FWIW, it appears that readr's `col_spec` doesn't seem to have an interaction model there at all:

```
> foo <- readr::cols(a = readr::col_integer())
> foo$a
NULL
> foo
cols(
  a = col_integer()
)
> foo["a"]
$<NA>
NULL
```

Reply:
Yeah unfortunately I think it has to be Field here, but in the <- methods I tried to support both ways of assigning.